app/controllers/course/controller.rb

```diff
@@ -38,6 +38,17 @@ def current_component_host
                                                           self)
   end
 
+  # Given an achievement, returns all valid achievement conditions it may depend on.
+  #
+  # @param [Course::Achievement] achievement The achievement for which to return conditions.
+  # @return [Array<Course::Achievement>]
+  def valid_achievement_conditions(achievement)
+    this_achievement = [achievement]
+    dependent_on = achievement.achievement_conditions
+    achievement.course.achievements - this_achievement - dependent_on
+  end
+  helper_method :valid_achievement_conditions
```

Comment:
hang on, I just noticed. Why is this in the course controller? shouldn't this be a model method?